src/model/users/activity.rs

```diff
@@ -0,0 +1,190 @@
+use crate::model::{Color, PerfType, Request};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, ops::Range};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Interval {
+    pub from: u64,
+    pub to: u64,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct RatingProgress {
+    pub before: u32,
+    pub after: u32,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Score {
+    pub win: u32,
+    pub loss: u32,
+    pub draw: u32,
+
+    #[serde(rename = "rp")]
+    pub rating_progress: RatingProgress,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Puzzles {
+    pub score: Score,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Followers {
+    pub ids: Vec<String>,
+
+    #[serde(rename = "nb", skip_serializing_if = "Option::is_none")]
+    pub count: Option<u32>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Follows {
+    #[serde(rename = "in")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gained: Option<Followers>,
+
+    #[serde(rename = "out")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub lost: Option<Followers>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CorrsepondenceOpponent {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub user: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rating: Option<u32>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde_with::skip_serializing_none]
+pub struct CorrespondenceGame {
+    pub id: String,
+    pub color: Color,
+    pub url: String,
+    pub opponent: CorrsepondenceOpponent,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
These would seem to be redundant, as we have `skip_serializing_none`. As a general rule, try and use `skip_serializing_none` where possible. The `lila` server is built to not return null unless absolutely necessary, so this is the right choice 99% of the time.